WordPress/Classes/ViewRelated/Gutenberg/EditorMediaUtility.swift

```diff
@@ -3,13 +3,18 @@ import Aztec
 import Gridicons
 
 final class AuthenticatedImageDownload: AsyncOperation {
+    enum DownloadError: Error {
+        case blogNotFound
+    }
+
     let url: URL
     let blogObjectID: NSManagedObjectID
     private let callbackQueue: DispatchQueue
     private let onSuccess: (UIImage) -> ()
     private let onFailure: (Error) -> ()
 
     init(url: URL, blogObjectID: NSManagedObjectID, callbackQueue: DispatchQueue, onSuccess: @escaping (UIImage) -> (), onFailure: @escaping (Error) -> ()) {
+        assert(!blogObjectID.isTemporaryID)
```

Comment:
For reference, since I always forget how `assert`, `fatalError`, etc behave in Release, so I checked [the docs](https://developer.apple.com/documentation/swift/assert(_:_:file:line:)) and verified this will be ignored in Release builds:

> Use this function for internal sanity checks that are active during testing but do not impact performance of shipping code. To check for invalid usage in Release builds, see `precondition(_:_:file:line:)`.

I think that's the kind of behavior we want. Fail ASAP during development to shorten the feedback cycle, but be graceful (`throw DownloadError.blogNotFound`) in Release builds.